ui/src/shared/components/Notification.js

```diff
@@ -0,0 +1,93 @@
+import React, {Component, PropTypes} from 'react'
+import {connect} from 'react-redux'
+import {bindActionCreators} from 'redux'
+
+import classnames from 'classnames'
+
+import {dismissNotification as dismissNotificationAction} from 'shared/actions/notifications'
+
+class Notification extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      opacity: 1,
+      height: 0,
+      dismiss: false,
+    }
+  }
+
+  componentDidMount = () => {
+    const {notification: {duration}} = this.props
+
+    // Trigger animation in
+    const {height} = this.notificationRef.getBoundingClientRect()
+    this.setState({height})
+
+    if (duration >= 0) {
+      // Automatically dismiss notification after duration prop
+      this.dismissTimer = setTimeout(this.handleDismiss, duration)
+    }
+  }
+
+  componentWillUnmount = () => {
```

Comment:
not sure this should be an assignment but rather just a declaration. why is this pattern used here? `this` should already be bounced in the lifecycle methods. the pattern we've used before for timers is to track the id of timers on component state, and use function declaration syntax, and then `clearTimeout(this.state.<id>)`. i don't think i'm a big fan of storing these timers on the class object.